src/safeds/data/tabular/containers/_table.py

```diff
@@ -1708,6 +1708,65 @@ def inverse_transform_table(self, fitted_transformer: InvertibleTableTransformer
         """
         return fitted_transformer.inverse_transform(self)
 
+    def join(
+        self,
+        right_table: Table,
+        left_names: str | list[str],
+        right_names: str | list[str],
+        *,
+        mode: Literal["inner", "left", "outer"] = "inner",
+    ) -> Table:
+        """
+        Join a table with the current table and return the result.
+
+        Parameters
+        ----------
+        right_table: Table
```

Comment:
```suggestion
        right_table:
```

Types in the docstrings are redundant, since we already have them in a more formalized manner in the code.